src/formatDate.js

```diff
@@ -7,8 +7,31 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateString, fromFormat, toFormat) {
+  const dateArray = dateString.split(fromFormat[3]);
+  const object = {};
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    object[fromFormat[i]] = dateArray[i];
+  }
+
+  if (object.YYYY) {
+    object.YY = object.YYYY.slice(2);
+  }
+
+  if (object.YY < 30) {
+    object.YYYY = '20' + `${object.YY}`;
+  } else {
+    object.YYYY = '19' + `${object.YY}`;
+  }
```

Comment:
The logic for converting years is incorrect. It should only convert from 'YYYY' to 'YY' or from 'YY' to 'YYYY' based on the `toFormat` array. Currently, it always assumes a conversion to 'YY' and then back to 'YYYY', which is not always the case. You need to adjust the logic to check the `toFormat` to determine if a conversion is necessary and perform the appropriate action.